pkg/plugins/resources/k8s/caching_converter.go

```diff
@@ -40,17 +40,6 @@ func (c *cachingConverter) ToCoreResource(obj k8s_model.KubernetesObject, out co
 		obj.GetResourceVersion(),
 		obj.GetObjectKind().GroupVersionKind().String(),
 	}, ":")
-	if obj.GetResourceVersion() == "" {
```

Comment:
to avoid repetition of this block
```
if out.Descriptor().HasStatus {
		status, err := obj.GetStatus()
		if err != nil {
			return err
		}
		if err := out.SetStatus(status); err != nil {
			return err
		}
	}
```

I also think there was a mistake in this logic - the lack of return in the `if obj.GetResourceVersion() == "" {`.